{,nixos/}pocket-id: 0.53.0 -> 1.1.0

gepbird opened the pull request:
Continuation of https://github.com/NixOS/nixpkgs/pull/410768

## https://github.com/pocket-id/pocket-id/releases/tag/v1.0.0

This contains breaking changes, see https://pocket-id.org/docs/setup/migrate-to-v1/.

The frontend now generates only static files and no longer includes a
binary for serving them. The backend has taken over the responsibility
of serving the static assets.

## https://github.com/pocket-id/pocket-id/releases/tag/v1.1.0

This introduces some lightweight analytics, I added an option for the NixOS module for discoverability.


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

gepbird committed:
pocket-id: match upstream go build config

gepbird committed:
nixos/pocket-id: refactor

ymstnt commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 411229`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>pocket-id</li>
  </ul>
</details>

Pyxels reviewed (approved):
basic usage with my typical workflows work!

gepbird committed:
{,nixos/}pocket-id: 0.53.0 -> 1.0.0

This contains breaking changes, see https://pocket-id.org/docs/setup/migrate-to-v1/.

The frontend now generates only static files and no longer includes a
binary for serving them. The backend has taken over the responsibility
of serving the static assets.

Co-authored-by: ymstnt <21342713+YMSTNT@users.noreply.github.com>

gepbird committed:
{,nixos/}pocket-id: make pocket-id-backend subpackage the main package

gepbird committed:
pocket-id: 1.0.0 -> 1.1.0

gepbird commented:
> We noticed an issue with the image loading like the logo and background falling back to their alt text due to 404 responses For now I'm rolling back from 1.1.0 to 1.0.0 and will figure out the reason when I'm less busy

Upstream bug report: https://github.com/pocket-id/pocket-id/issues/597

It's also present on 1.0.0, so the 1.1.0 update didn't cause it. After all I don't think it's a blocker


> I got this working as a first time user of pocket-id, however I was unable to add passkeys when accessing the app through http://localhost:1411/login/setup as documented at the bottom of the [pocket-id installation instructions](https://pocket-id.org/docs/setup/installation). I was able to add passkeys after setting up a reverse proxy with https and setting APP_URL and TRUST_PROXY

@jmoggr It may be intentional by the Pocket ID software to make it more secure by only allowing it with HTTPS. I don't think it's a bug with the nix package or module, but we haven't looked into it deeply. I think It's worth reporting it at https://github.com/pocket-id/pocket-id/issues

gepbird committed:
nixos/pocket-id: add option to disable analytics

Co-authored-by: ymstnt <21342713+YMSTNT@users.noreply.github.com>

marcusramberg commented:
lgtm :)

gepbird commented:
@jmoggr Now the example env file (and looks like the installation-instruction docs that you linked) contains https instead of http  from https://github.com/pocket-id/pocket-id/commit/c24a5546a5254d56f58658a2d3d74b5431508b67. If this was your issue, it's fixed in the docs. I haven't checked the setup process
